pkg/tsdb/azuremonitor/azuremonitor.go

```diff
@@ -161,3 +164,179 @@ func (s *Service) newQueryMux() *datasource.QueryTypeMux {
+
+	if defaultWorkspaceId == "" {
+		workspacesUrl := fmt.Sprintf("%v/subscriptions/%v/providers/Microsoft.OperationalInsights/workspaces?api-version=2017-04-26-preview", dsInfo.Routes["Azure Monitor"].URL, dsInfo.Settings.SubscriptionId)
+		workspacesReq, err := http.NewRequest(http.MethodGet, workspacesUrl, nil)
+		if err != nil {
+			return nil, err
+		}
+		res, err := dsInfo.Services["Azure Monitor"].HTTPClient.Do(workspacesReq)
+		if err != nil {
+			return nil, err
+		}
+		var target struct {
+			Value []types.LogAnalyticsWorkspaceResponse
+		}
+		err = json.NewDecoder(res.Body).Decode(&target)
+		if err != nil {
+			return nil, err
+		}
+
+		if len(target.Value) == 0 {
+			return nil, errors.New("no default workspace found")
+		}
+		defaultWorkspaceId = target.Value[0].Properties.CustomerId
+	}
+
+	body, err := json.Marshal(map[string]interface{}{
+		"query": "AzureActivity | limit 1",
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	workspaceUrl := fmt.Sprintf("%v/v1/workspaces/%v/query", dsInfo.Routes["Azure Log Analytics"].URL, defaultWorkspaceId)
+	workspaceReq, err := http.NewRequest(http.MethodPost, workspaceUrl, bytes.NewBuffer(body))
+	workspaceReq.Header.Set("Content-Type", "application/json")
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := dsInfo.Services["Azure Log Analytics"].HTTPClient.Do(workspaceReq)
+	if err != nil {
+		return nil, err
+	}
+
+	return res, nil
+}
+
+func checkAzureMonitorGraphHealth(dsInfo types.DatasourceInfo) (*http.Response, error) {
+	body, err := json.Marshal(map[string]interface{}{
+		"query":         "Resources | project id | limit 1",
+		"subscriptions": []string{dsInfo.Settings.SubscriptionId},
+	})
+	if err != nil {
+		return nil, err
+	}
+	url := fmt.Sprintf("%v/providers/Microsoft.ResourceGraph/resources?api-version=2021-06-01-preview", dsInfo.Routes["Azure Resource Graph"].URL)
+	request, err := http.NewRequest(http.MethodPost, url, bytes.NewBuffer(body))
+	request.Header.Set("Content-Type", "application/json")
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := dsInfo.Services["Azure Resource Graph"].HTTPClient.Do(request)
+	if err != nil {
+		return nil, err
+	}
+
+	return res, nil
+}
+
+func (s *Service) CheckHealth(ctx context.Context, req *backend.CheckHealthRequest) (*backend.CheckHealthResult, error) {
+	dsInfo, err := s.getDSInfo(req.PluginContext)
+	if err != nil {
+		return nil, err
+	}
+
+	status := backend.HealthStatusOk
+	metricsLog := "Successfully connected to Azure Monitor endpoint."
+	logAnalyticsLog := "Successfully connected to Azure Log Analytics endpoint."
+	graphLog := "Successfully connected to Azure Resource Graph endpoint."
+
+	metricsRes, err := checkAzureMonitorMetricsHealth(dsInfo)
+	if err != nil || metricsRes.StatusCode != 200 {
+		if err != nil {
+			return nil, err
+		} else {
+			body, err := io.ReadAll(metricsRes.Body)
+			fmt.Println("here", body, err)
+			if err != nil {
+				return nil, err
+			}
+			backend.Logger.Error(string(body))
```

Comment:
> * In case anything sensitive is exposed

that was my point above, we are just using the credentials that the user is giving us to create a request that they could create themselves. If there is sensitive information, they have access to that information anyway, there is no point on hiding it.

> The Azure errors can be quite verbose and I don't think this will always present very well in the UI. Happy to have a discussion about this.

I'd rather have a verbose error than no error at all so you don't know what's happening. If the user wants to fix the error, they need to access the Grafana logs and looks to the verbose error anyway, which is even more difficult.

If you still want to discuss this we can jump into a call!